common/djangoapps/third_party_auth/models.py

```diff
@@ -418,3 +421,47 @@ def current(cls, entity_id):
 
         cache.set(cls.cache_key_name(entity_id), current, cls.cache_timeout)
         return current
+
+
+class LTIProviderConfig(ProviderConfig):
+    """
+    Configuration required for this edX instance to act as a LTI
+    Tool Provider and allow users to authenticate and be enrolled in a
+    course via third party LTI Tool Consumers.
+    """
+    prefix = 'lti'
+    backend_name = 'lti'
+    icon_class = None  # This provider is not visible to users
+    accepts_logins = False  # LTI login cannot be initiated by the tool provider
+    KEY_FIELDS = ('lti_consumer_key', )
+
+    lti_consumer_key = models.CharField(
+        max_length=255,
+        help_text=(
+            'The name that the LTI Tool Consumer will use to identify itself'
+        )
+    )
+    lti_consumer_secret = models.CharField(
+        max_length=255,
+        help_text=(
+            'The shared secret that the LTI Tool Consumer will use to '
+            'authenticate requests. Only this edX instance and this '
+            'tool consumer instance should know this value.'
+        )
```

Comment:
Is this usually generated when setting up the LTI provider, or the consumer? If the former, it would be nice for users and likely encourage better security practices if you give this a callable default that will generate a secure random value for the user, when first creating a new configuration. [The LMS Oauth2 provider already does that and provides a nice example.](https://github.com/caffeinehit/django-oauth2-provider/blob/7fb926f3c81457/provider/oauth2/models.py#L45)